client/state/user-settings/actions.js

```diff
@@ -81,6 +83,27 @@ export const removeUnsavedUserSetting = ( settingName ) => ( {
 	settingName,
 } );
 
+/**
+ * Checks if an incoming change to settings.language is a change to the existing settings
+ * Currently the assumption is that if a settings.locale_variant slug exists, then that is the current language
+ *
+ * @param  {string}  languageSettingValue the newly-set language slug string.
+ * @param  {object}  settings user settings object.
+ * @returns {boolean} if the language setting has been changed.
+ */
+function hasLanguageChanged( languageSettingValue, settings = {} ) {
```

Comment:
I wonder if we could create a separate PR with the new redux functionality. It would definitely help to keep this one a bit smaller.